conan_provider.cmake

```diff
@@ -76,6 +76,8 @@ function(detect_arch ARCH)
         set(_ARCH x86)
     elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "AMD64|amd64|x86_64" OR CMAKE_OSX_ARCHITECTURES MATCHES x86_64)
         set(_ARCH x86_64)
+    else()
+        set(_ARCH "${CMAKE_SYSTEM_PROCESSOR}")
```

Comment:
This can be some architecture not defined in Conan, which would give an error saying so. I guess it is better than not defining the architecture at all.

If this is the only functional change, better treat it separately in a different PR.

Reply:
Yes, that's the idea. I have a completely non-standard architecture, which requires a custom `settings.yml` to be installed. The best solution seems to be to let cmake-conan pass it through directly in that case.

I'll update the commit message to add more detail.